src/main/java/io/lumigo/core/utils/AwsUtils.java

```diff
@@ -243,6 +243,24 @@ public static synchronized Span.READINESS getFunctionReadiness() {
         }
     }
 
+    public static int getJavaVersion() {
+        try {
+            String version = System.getProperty("java.version");
+            if (version.startsWith("1.")) {
+                version = version.substring(2, 3);
+            } else {
+                int dot = version.indexOf(".");
+                if (dot != -1) {
+                    version = version.substring(0, dot);
+                }
+            }
+            return Integer.parseInt(version);
+        } catch (Exception e) {
+            Logger.error("Failed to parse java version", e);
+            return -1;
```

Comment:
maybe the fallback should be the new versions. more likely to fail to parse new versions...